designs/2024-deprecated-rule-metadata/README.md

```diff
@@ -0,0 +1,262 @@
+
+## Alternatives
+
+<!--
+    What other designs did you consider? Why did you decide against those?
+
+    This section should also include prior art, such as whether similar
+    projects have already implemented a similar feature.
+-->
+
+### Do nothing
+
+This would leave the current `meta.deprecated` and `meta.replacedBy` properties as they are, which would continue to be ambiguous and limited in the information they can provide.
+
+### Create a new property
+
+Create a new property, e.g. `meta.deprecation`,
+
+## Open Questions
+
+<!--
+    This section is optional, but is suggested for a first draft.
+
+    What parts of this proposal are you unclear about? What do you
+    need to know before you can finalize this RFC?
+
+    List the questions that you'd like reviewers to focus on. When
+    you've received the answers and updated the design to reflect them,
+    you can remove this section.
+-->
+
+1. Is there additional deprecation information we'd like to represent? Note that additional information can always be added later, but it's good to consider any possible needs now.
+2. Should `meta.deprecated.plugin.name` accommodate different package registries (e.g. [jsr](https://jsr.io/) with `jsr:eslint-plugin-example`)
+  - If it is a concern, plugin developers can use a direct URL
+3. Which "extension points" (rules, processors, configurations, parsers, formatters) shold be supported?
+  - Focus on rules as other extension points have no well-defined use cases
+
+## Help Needed
+
+<!--
+    This section is optional.
+
+    Are you able to implement this RFC on your own? If not, what kind
+    of help would you need from the team?
+-->
+
+I should be able to handle the minimal changes needed in ESLint core, and can kick off some of the changes needed in community projects.
+
+## Frequently Asked Questions
+
+- Why not provide a property to describe how to migrate to the replacement rule which requires an option to be set?
+  - The options of the replacement rule could change and it is unlikely that a deprecated rules gets updated to accommodate the change
+
+<!--
+    This section is optional but suggested.
+
+    Try to anticipate points of clarification that might be needed by
+    the people reviewing this RFC. Include those questions and answers
+    in this section.
+-->
+
+## Related Discussions
+
+<!--
+    This section is optional but suggested.
+
+    If there is an issue, pull request, or other URL that provides useful
+    context for this proposal, please include those links here.
+-->
+
+- [The original RFC](https://github.com/eslint/rfcs/pull/116)
+- [The issue triggering this RFC](https://github.com/eslint/eslint/issues/18061)
+- Inspirations
+  - <https://github.com/jsx-eslint/eslint-plugin-react/pull/3469#discussion_r1002316631>
+  - <https://github.com/eslint/eslint/issues/5774#issuecomment-220640368>
+- Related
+  - <https://github.com/eslint/eslint/issues/17842>
+  - <https://github.com/eslint/eslint/issues/18694>
```

Comment:
```suggestion
```

I think this isn't related. Frozen rules are not deprecated.

Reply:
I added this reference for `availableUntil: null` which should signify that a rule is frozen.

Reply:
The linked issue is only about marking some non-deprecated rules as frozen, so I still think it isn't related to the subject of this RFC.

frozen = we're not accepting new features (e.g., new options), but the rule is actively maintained (bug fixes & support for new syntax).

deprecated = we're not making any changes, and we don't guarantee that the rule won't be removed in a future major version:

https://eslint.org/docs/latest/use/rule-deprecation